doc/application/index.rst

```diff
@@ -235,6 +235,133 @@ Follow these steps to create a new application directory. (Refer to
    ``source zephyr-env.sh`` or execute ``zephyr-env.cmd`` before building the
    application.
 
+
+.. _env_vars:
+
+Setting Variables
```

Comment:
I am a bit unsure of moving the env vars section to the application development guide, where is the duplication you mention in the commit message?

Reply:
`Important Build System Variables` for example.

Reply:
> I am a bit unsure of moving the env vars section to the application development guide,

we can put somewhere else, but this does not need its own standalone page.

Reply:
> Important Build System Variables for example.

I just noticed this PR and I really disagree. I wrote the environment variable page and this change is confusing and a step backwards.  "Important build system variables" is a list of **cmake** variables, **not** environment variables. You are obscuring this important difference by stuffing a section simply titled "Setting variables" in this page. I'll send a logical revert.